reconcile/deadmanssnitch.py

```diff
@@ -147,41 +169,22 @@ def get_current_state(
     def get_desired_state(
         self,
         clusters: list[ClusterV1],
-        current_state: dict[str, Snitch],
         cluster_to_prometheus_mapping: dict[str, str],
-    ) -> dict[str, Snitch]:
+    ) -> dict[str, ClusterFields]:
         desired_state = {
-            cluster.name: self.map_desired_snitch_value(
-                cluster.name, current_state, cluster_to_prometheus_mapping
+            cluster.name: ClusterFields(
+                prometheus_url=cluster_to_prometheus_mapping.get(cluster.name),
```

Comment:
`cluster_to_prometheus_mapping` seems not needed here, as the url is from `cluster.prometheus_url.replace("https://", "")`, since we already have `cluster` here, we don't need to use `cluster_to_prometheus_mapping`. Can use a simple selector function `get_snitch_name(cluster)`.

And if you check the usage of `cluster_to_prometheus_mapping` in `get_current_state`, it's using reverse lookup, if create the reverse mapping directly

```python
snitch_name_to_cluster_name_mapping = {get_snitch_name(cluster): cluster.name for cluster in clusters}
```

Then you'll find it's easier to construct current state, no need to do O(N^2) iterate on snitches and clusters, can directly construct

```python
current_state = {
    cluster_name: self.add_vault_data(cluster.name, snitch, snitch_secret_path)
    for snitch in snitches
    if (cluster_name := snitch_name_to_cluster_name_mapping.get(snitch.name))
}

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4238/files/d7c7fbc0e3669589f076bee03e7c769137c7bd9b..c1c19294fa3b76f8bd31cfe02ec5dd0e9f545016#diff-27ae2e4c7d059f24158bac19d459e906eb2890c078c3d79eb0205423f5ae5e6cR162-R171

here no need to build `snitches_with_cluster_mapping` first then build `current_state`, given the list of snitches and `snitch_name_to_cluster_name_mapping`, can build `current_state` directly.